src/client.ts

```diff
@@ -27,73 +33,160 @@ import type { Browsers } from "./types.ts";
  *
  * @example
  * ```js
- * const client = await Client.create([
+ * const { browser, page } = await Client.create([
  *   "chrome", "--headless",
  * ], {
  *   hostname: "localhost",
  *   port: 1234
  * }, "chrome", undefined);
- * const page = await client.goTo("https://drash.land");
  * ```
  */
 export class Client {
-  #protocol: ProtocolClass;
+  /**
+   * Whilst we don't be using this like a page would, it is used
+   * as a 'general' protocol, not specific to pages, but maybe
+   * to get targets, or general information of the overal browser
+   */
+  readonly #protocol: ProtocolClass;
+
+  /**
+   * The sub process that runs headless chrome
+   */
+  readonly #browser_process: Deno.Process;
+
+  /**
+   * Track if we've closed the sub process, so we dont try close it when it already has been
+   */
+  #browser_process_closed = false;
+
+  /**
+   * What browser we running?
+   */
+  readonly browser: Browsers;
+
+  pages: Page[] = [];
```

Comment:
can we make it private and use a method to getPage(index) ? and handle outOfBound exception?

Reply:
we can do the outofbound tstuff, but i think we need it public so other classes can access it

Reply:
actually ive managed to do both: https://github.com/drashland/sinco/pull/103/commits/96b934689e3e2a4a68289aadc3032128c6e53002